src/global/utils/auth.tsx

```diff
@@ -16,34 +16,71 @@
  * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
  * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
+'use client';
 
-import { createContext, Dispatch, ReactNode, SetStateAction, useState, useContext } from 'react';
+import {
+	createContext,
+	Dispatch,
+	ReactNode,
+	SetStateAction,
+	Suspense,
+	useContext,
+	useEffect,
+	useState,
+} from 'react';
 import Cookies from 'js-cookie';
-import { EGO_JWT_KEY } from './constants';
+import { usePathname } from 'next/navigation';
+import Header from '@/app/components/Header';
+import { DnaLoader } from '@icgc-argo/uikit';
+import { EGO_JWT_KEY } from '../constants';
 
 type AuthContextValue = {
 	egoJwt: string;
 	setEgoJwt: Dispatch<SetStateAction<string>>;
+	loggingIn: boolean;
+	setLoggingIn: Dispatch<SetStateAction<boolean>>;
 };
 
 const AuthContext = createContext<AuthContextValue>({
 	egoJwt: '',
 	setEgoJwt: () => '',
+	loggingIn: false,
+	setLoggingIn: () => false,
 });
 
-export const useAuthContext = () => useContext(AuthContext);
+export const getStoredToken = () => Cookies.get(EGO_JWT_KEY);
 
-const removeToken = () => {
-	Cookies.remove(EGO_JWT_KEY);
+export const storeToken = (egoToken: string) => {
+	Cookies.set(EGO_JWT_KEY, egoToken);
 };
 
 export const logOut = () => {
-	removeToken();
+	Cookies.remove(EGO_JWT_KEY);
 };
 
-export default function AuthProvider({ children }: { children: ReactNode }) {
-	const storedToken = Cookies.get(EGO_JWT_KEY);
+export function AuthProvider({ children }: { children: ReactNode }) {
+	const storedToken = getStoredToken();
 	const [egoJwt, setEgoJwt] = useState(storedToken || '');
-	const authData = { egoJwt, setEgoJwt };
-	return <AuthContext.Provider value={authData}>{children}</AuthContext.Provider>;
+	const path = usePathname();
+	const initLoginState = path === '/logging-in' && !egoJwt.length ? true : false;
```

Comment:
I think @justincorrigible comment is good about changing the name because init does seem like initializing

```javascript
!egoJwt.length ?  true : false
```
this can just be the conditional `!egoJwt.legnth` - whenever a ternary is just returning `conditional ?  true : false` you can just do the conditional if that makes sense? Otherwise it's just if `true return true` if `false return false`